import { PageError } from "../../../shared/errors";
import { Page } from "../../../shared/schemas";

export async function DeletePage(id: string) {
    try {
        const page = await Page.findOne({where: {id}}).catch(_error => {throw new PageError('Ha ocurrido un error al revisar la p치gina.')}).then(page => page);
        if (!page) throw new PageError(`La p치gina con ID ${id} no existe.`);
        await page.destroy().catch(_error => {throw new PageError('Ha ocurrido un error al tratar de eliminar la p치gina.')});
    } catch (error) {
        if (error instanceof Error && error.message) throw new PageError(error.message);
        else throw new Error('Ha ocurrido un error al eliminar la p치gina.');
    }
}